modules/core/plugin-container/src/main/java/org/rhq/core/pc/inventory/InventoryManager.java

```diff
@@ -1928,24 +1931,24 @@ private Configuration mergeExistingResourcePluginConfiguration(Resource resource
 
         // If there is no update necessary just return the current plugin config of the existing resource
         Configuration result = resource.getPluginConfiguration();
-
         ConfigurationDefinition configDef = resource.getResourceType().getPluginConfigurationDefinition();
         if (null == configDef) {
             return result;
         }
-
+        Map<String, PropertyDefinition> propDefs = configDef.getPropertyDefinitions();
         Configuration existingPluginConfig = resource.getPluginConfiguration().deepCopy(false);
-        Configuration defaultPluginConfig = ConfigurationUtility.createDefaultConfiguration(configDef);
+
         boolean configChanged = false;
 
         // for each property, update the existing plugin config if discovery has set a non-default value
         for (String propertyName : pluginConfig.getAllProperties().keySet()) {
             Property discoveredProp = pluginConfig.get(propertyName);
-            Property defaultProp = defaultPluginConfig.get(propertyName);
-            if (!discoveredProp.equals(defaultProp)) {
+            // Only overrides the property if it is a read-only property
+            // See BZ: 1379834
+            if (propDefs.get(propertyName).isReadOnly()) {
```

Comment:
This looks well but I'm not sure how this will affect other plugins.

Maybe we could define an attribute on the XML. Something like `override_strategy` that let you choose how to globally override the attribute values, DEFAULT, READ_ONLY, ONLY_NOT_NULL_DEFAULT, etc.

Or maybe i'm just being paranoid and no one cares/uses this.